lib/open_api_spex/cast/discriminator.ex

```diff
@@ -93,7 +94,11 @@ defmodule OpenApiSpex.Cast.Discriminator do
     end
   end
 
-  defp find_discriminator_schema(discriminator, _, schemas) do
+  defp find_discriminator_schema(discriminator, _, schemas) when is_atom(discriminator) do
+    Enum.find(schemas, &Kernel.==(&1."x-struct", discriminator))
```

Comment:
What happens when a schema is set to `struct?: false`?

Reply:
Addressed in latest commit.